org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/runtime/VMInputStream.java

```diff
@@ -57,14 +57,16 @@ public boolean markSupported() {
 }
 @Override
 public int read() throws IOException {
-	try {
-		return this.input.read();
-	} catch (IOException e) {
-		if (isRunning()) {
-			return read();
-		}
-		throw e;
+	while (true) {
```

Comment:
So this would be an endless loop instead of Stackoverflow? Doesn't sound better, but yes the recursion seems wrong too.